### DISCUSSION TITLE
How to make movement some components in a iter one by one?

### DISCUSSION THREAD

**Discussion by JiatLn** [bevyengine/bevy#9817] [Q&A]
```rust
pub fn monsters_movement_system(
    mut monsters_query: Query<&mut Transform, With<Monster>>,
) {
    for mut monster_transform in monsters_query.iter_mut() {
        let mut direction = Vec3::ZERO;
        direction += Vec3::new(-1.0, 0.0, 0.0);
        let target = monster_transform.translation + direction;
        monster_transform.translation = target;
    }
}
```

I have some code above, and how i make the monster movement one by one, now it movement together in a frame

**Comment by laundmo** [comment]
Each system runs once per frame. As this one loops over all monsters, it will modify all monsters once per frame.
I think the easiest solution if you want staggered movement (one monster moves each second) would be to move the monsters on a timer: either a single one which times however long each monster is moved and stores which is the current monster, or a timer per monster which stores when that monster should move next.
